src/Storages/KeyDescriptionMonotonicityChecker.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <Functions/IFunction.h>
+#include <Parsers/ASTFunction.h>
+#include <Storages/KeyDescription.h>
+#include <Interpreters/Context_fwd.h>
+#include <Core/Field.h>
+#include <Storages/MergeTree/RPNBuilder.h>
```

Comment:
Would it be possible to use forward declarations, if applicable (for the pointer and reference types)?